src/spectrum.rs

```diff
@@ -0,0 +1,375 @@
+
+
+use std::fmt;
+use std::ops::{Add, Sub, Mul, Div};
+
+use errors::*;
+use flt;
+use num::{Float, FromPrimitive, Zero};
+use ordered_float::OrderedFloat;
+use white_point::WhitePoint;
+use xyz::Xyz;
+
+/// The smallest wavelength represented in the `Spectrum`
+/// data structure. This is the first value in the
+/// map's wavelength.
+pub const SPECTRUM_MIN_LAMBDA: f32 = 360.0;
+
+/// The largest wavelength represented in the `Spectrum`
+/// data structure. This is the last value in the
+/// map's wavelength.
+#[allow(dead_code)]
+pub const SPECTRUM_MAX_LAMBDA: f32 = 830.0;
+
+pub const SPECTRUM_SAMPLES: usize = 95;
+const SPECTRUM_SAMPLE_STEP: usize = 5;
```

Comment:
I've refactored the constants to use constant expressions and removed the redundant 'SPECTRUM' naming (as this is obvious given we're in the 'spectrum.rs' module).